pubsublite-spark-sql-streaming/src/main/java/com/google/cloud/pubsublite/spark/PslContinuousInputPartition.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsublite.spark;
+
+import com.google.cloud.pubsublite.Offset;
+import com.google.cloud.pubsublite.SequencedMessage;
+import com.google.cloud.pubsublite.SubscriptionPath;
+import com.google.cloud.pubsublite.internal.BufferingPullSubscriber;
+import com.google.cloud.pubsublite.internal.CheckedApiException;
+import com.google.cloud.pubsublite.internal.PullSubscriber;
+import com.google.cloud.pubsublite.internal.wire.PubsubContext;
+import com.google.cloud.pubsublite.internal.wire.SubscriberBuilder;
+import com.google.cloud.pubsublite.proto.Cursor;
+import com.google.cloud.pubsublite.proto.SeekRequest;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.flogger.GoogleLogger;
+import java.io.Serializable;
+import java.util.concurrent.*;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.sources.v2.reader.ContinuousInputPartition;
+import org.apache.spark.sql.sources.v2.reader.InputPartitionReader;
+import org.apache.spark.sql.sources.v2.reader.streaming.ContinuousInputPartitionReader;
+import org.apache.spark.sql.sources.v2.reader.streaming.PartitionOffset;
+
+public class PslContinuousInputPartition
+    implements ContinuousInputPartition<InternalRow>, Serializable {
+  private static final GoogleLogger log = GoogleLogger.forEnclosingClass();
+
+  private final PslPartitionOffset startOffset;
+  private final PslDataSourceOptions options;
+
+  public PslContinuousInputPartition(PslPartitionOffset startOffset, PslDataSourceOptions options) {
+    this.startOffset = startOffset;
+    this.options = options;
+  }
+
+  @VisibleForTesting
+  public static ContinuousInputPartitionReader<InternalRow> createPartitionReader(
+      SubscriptionPath subscriptionPath,
+      PslPartitionOffset currentOffset,
+      PullSubscriber<SequencedMessage> subscriber,
+      ScheduledExecutorService pullExecutorService) {
+    return new PslContinuousInputPartitionReader(
+        subscriptionPath, currentOffset, subscriber, pullExecutorService);
+  }
+
+  @Override
+  public InputPartitionReader<InternalRow> createContinuousReader(PartitionOffset offset) {
+    assert PslPartitionOffset.class.isAssignableFrom(offset.getClass())
+        : "offset is not assignable to PslPartitionOffset";
+
+    PslPartitionOffset pslOffset = (PslPartitionOffset) offset;
+    PslPartitionOffset currentOffset =
```

Comment:
It's not a range, it's a point (unless you are talking about microbatch). I can see how this could be confusing and not well managed.
I am going to add a PartitionOffset from PSL side, and I will put some conversion utils that converts between pubsublite.PartitionOffset and PslPartitionOffset. Hope that would clear it up.

Reply:
Created https://github.com/googleapis/java-pubsublite/pull/404 and based this PR on top of it.